addon/-private/key-values.ts

```diff
@@ -0,0 +1,29 @@
+import isObject from '../utils/is-object';
+
+// 'foo.bar.zaz'
+let keysUpToValue: any[] = [];
```

Comment:
If this is used as an accumulator for recursion, this should be an argument to the function. Having it persisted across function calls can be error prone. If you don't want to expose the accumulator outside this module, you can use this common pattern:

```js
const myRecursiveFunctionWithAccumulator = (a, acc) => {
  /* do whatever recursive stuff you need to do */
};
export const myFunction = (a) => myRecursiveFunctionWithAccumulator(a, []);
```